js/index.js

```diff
@@ -1,81 +1,163 @@
-                <div class="search-result__anime-content__img">
-                    <img src="${imageUrl}" alt="image">
-                </div>
-                <div class="search-result__anime-content__bottom">
-                    <div class="search-result__anime-content__score">&#x2605 ${score}</div>
-                    <div class="search-result__anime-content__episodes">ep. ${episodeCount}</div>
-                </div>`
-    };
-
-    btn.addEventListener('click', function (e) {
-        e.preventDefault();
-        const searchVar = userSearch.value;
-        console.log(searchVar);
-        const loadingAnimation = document.querySelector('.letter-holder');
-        loadingAnimation.style = "display: block;";
-
-        fetch(`https://api.jikan.moe/v3/search/anime?q=${searchVar}`)
-            .then(function (resp) {
-                if (resp.ok) {
-                    return resp.json()
-                }
-
-                return resp.json().then(errorData => { 
-                    const e = new Error ('Something went wrong');
-                    e.data = errorData;
-                    throw e;
+  const userSearch = document.querySelector('.search-form__content__input');
+  const btn = document.querySelector('.search-form__content__btn');
+
+  function createItemHtml(result) {
+    const {
+      title,
+      score,
+      image_url: imageUrl,
+      episodes: episodeCount
+    } = result;
+    return `<div class="search-result__anime-content__title">${title}</div>
+          <div class="search-result__anime-content__img">
+            <img src="${imageUrl}" alt="image">
+          </div>
+          <div class="search-result__anime-content__bottom">
+            <div class="search-result__anime-content__score">&#x2605 ${score}</div>
+            <div class="search-result__anime-content__episodes">ep. ${episodeCount}</div>
+          </div>`
+  };
+
+  function animeInfoTemplate(className, info = '-') {
+    return `<div class="search-result__anime-content_specific-anime-${className}">${info}</div>`
+  };
+
+  function checkTitleExistence(title) {
+    if (title) {
+      return title;
+    }
+    return ''
+  };
+
+  function formatDate(dateWithWrongFormat) {
+    if (dateWithWrongFormat) {
+      const year = new Date(dateWithWrongFormat).getFullYear();
+      const mounth = new Date(dateWithWrongFormat).getMonth();
+      const day = new Date(dateWithWrongFormat).getDay();
+      return year + '.' +
+             (mounth > 9 ? mounth : String(mounth).padStart(2, '0')) + '.' +
+             (day > 9 ? day : String(day).padStart(2, '0'))
+    }
+    return '-'
+  };
+
+  function getNameListFromObjects(objects) {
+    if (objects && objects.length !== 0) {
+      return objects.map(o => o.name).join(', ');
+      // return objects.reduce((nameList, object, i) => {
+      //   const isLast = objects.length === i + 1 
+      //   return nameList + object.name + (isLast? '' : ', ');
+      // }, '');
```

Comment:
Please remove commented lines

Reply:
All your remarks have been corrected in commit that are below